### PR TITLE
French Translation JavaScript Basics n°1 and n°2

### PR DESCRIPTION
**Pull request by Tagada85**
French translation 
-Comment your Javascript code
-Declare Javascript variables

### CODE REVIEW

File: seed/challenges/01-front-end-development-certification/basic-javascript.json [hunk 1]
```diff
@@ -40,6 +40,19 @@
         "El comentario de barra-estrella-estrella-barra, comentará todo lo que haya entre los caracteres <code>/*</code> y <code>*/</code>:",
         "<code>/* Este es también un comentario */</code>",
         "Trata de crear uno de cada uno."
+      ],
+      "titleFr" : "Commentez votre code JavaScript",
+      "descriptionFr" : [
+        "Les commentaires sont des lignes de code que Javascript va intentionnellement ignorer. Les commentaires sont un bon moyen de laisser des notes à votre intention ou pour d'autres personnes qui devront comprendre ce que fait votre code.",
+        "Il y a two façons d'écrire des commentaires en JavaScript.",
```
Comment (mizujin) [inline seed/challenges/01-front-end-development-certification/basic-javascript.json]: Should read "Il y a **deux** façons.."

File: seed/challenges/01-front-end-development-certification/basic-javascript.json [hunk 2]
```diff
@@ -83,6 +96,20 @@
...
         "Ahora, utiliza la palabra clave <code>var</code> para crear una variable llamada <code>myName</code>. Establecele como valor tu nombre, entre comillas dobles. ",
         "Mira el ejemplo con <code>ourName</code> si te quedas atascado."
+      ],
+      "titleFr" : "Déclarer des variables en JavaScript",
+      "descriptionFr" : [
+        "En informatique, une <dfn>donnée</dfn> comporte tout ce qui peut avoir du sens pour l'ordinateur. Javascript dispose de sept <code>types de données</code> différents qui sont <code>undefined</code>, <code>null</code>, <code>boolean</code>, <code>string</code>, <code>symbol</code>, <code>number</code> et <code>object</code>.",
+        "Par exemple, les ordinateurs font la différence entre les nombres tel que <code>12</code> et les <code>strings</code>, tels que <code>\"12\"</code>, <code>\"dog\"</code>, ou <code>\"123 cats\"</code>, qui sont des collections de caractères. Les ordinateurs peuvent faire des opérations mathématiques sur un nombre, mais par sur une chaîne de caratères.",
```
Comment (mizujin) [inline seed/challenges/01-front-end-development-certification/basic-javascript.json]: Small typo on the end : "par sur une chaîne de caratères" should read par sur une chaîne de cara**c**tères

File: seed/challenges/01-front-end-development-certification/basic-javascript.json [hunk 3]
```diff
@@ -83,6 +96,20 @@
...
+        "Les <dfn>variables</dfn> permettent aux ordinateurs de stocker et manipuler les données de façon dynamique. Ils font cela en utilisant un <blockquote>label</blockquote> qui référence la donnée plutôt que d'utiliser la donnée elle-même. Les sept types de données peuvent être stockés dans une variable.",
```
Comment (mizujin) [inline seed/challenges/01-front-end-development-certification/basic-javascript.json]: This :  un <blockquote>label</blockquote> **qui référence** la donnée.. Could be :  un <blockquote>label</blockquote> **qui réfère** à la donnée..

File: seed/challenges/01-front-end-development-certification/basic-javascript.json [hunk 4]
```diff
@@ -83,6 +96,20 @@
...
+        "Les <code>variables</code> sont similaires aux x et y que vous utilisez en mathématiques, ce qui veut dire que ce sont des noms simples qui représentent les données que l'on veut utiliser. Les <code>variables</code> de l'ordinateur diffèrent de celles des mathématiques car elles peuvent stocker différentes valeurs à différents moments.",
+        "On peut dire à JavaScript de créer ou <code>déclarer</code> une variable en la précédant du mot-clé <code>var</code>, comme ceci:",
+        "<code>var notreNom</code>;",
+        "crée une <code>variable</code> appelée <code>notreNom</code>.On termine nos expressions avec un point virgule en JavaScript.",
+        "Le nom d'une <code>variable</code> peut être composé de nombres, lettres et <code>$</code> ou <code>_</code>, mais ne peut contenir d'espaces ou commencer par un nombre.",
+        "<h4>Instructions</h4>",
+        "Utilisez le mot-clé <code>var</code> pour créer une variable appelée <code>monNom</code>",
+        "<strong>Hint</strong><br>Regardez à l'exemple de <code>notreNom</code> si vous vous êtes bloqué."
```
Comment (mizujin) [inline seed/challenges/01-front-end-development-certification/basic-javascript.json]: <strong>Hint</strong><br>Regardez... Should read <strong>Conseil</strong><br>Regardez...

**Comment by bugron** [comment]
@Tagada85 any progress fixing this PR?

**Comment by QuincyLarson** [comment]
@Tagada85 awesome work with this translation. Once you fix this we will be excited to merge it. We will have French language support on Free Code Camp very soon, so people will start using these translations.

**Comment by mizujin** [comment]
@bugron Is there any way I can fix the file and PR here ?

**Comment by bugron** [comment]
@mizujin The best way I can think of (not sure it'll work in for this PR though) is to rebase this branch no top of the latest `staging` branch and squash your two last commits into one. But there is an easier way. Just close this PR and open a new one where you have only one commit. Make sure to fetch latest `staging` branch (`git pull origin staging --rebase)` and then create your branch on top of latest `staging`.

**Comment by raisedadead** [comment]
Closing this in favor of https://github.com/FreeCodeCamp/FreeCodeCamp/pull/9815

**Pull request closed without merging**